codalab/worker/local_run/docker_image_manager.py

```diff
@@ -136,6 +136,32 @@ def get(self, image_spec):
         :param image_spec: Repo image_spec of docker image being requested
         :returns: A DockerAvailabilityState object with the state of the docker image
         """
+
+        def get_local_image(image_spec, success_message, failure_message):
+            """
+            Try to get the image specified by image_spec from local. 
+            Return ImageAvailabilityState.
+            """
+            try:
+                image = self._docker.images.get(image_spec)
+                digests = image.attrs.get('RepoDigests', [image_spec])
+                digest = digests[0] if len(digests) > 0 else None
+                with self._lock:
+                    self._image_cache[digest] = ImageCacheEntry(
+                        id=image.id,
+                        digest=digest,
+                        last_used=time.time(),
+                        virtual_size=image.attrs['VirtualSize'],
+                        marginal_size=image.attrs['Size'],
+                    )
+                return ImageAvailabilityState(
+                    digest=digest, stage=DependencyStage.READY, message=success_message
+                )
+            except Exception:
```

Comment:
There are only two exceptions here based on documentation of [docker rest API:get(image_id) ](https://docker-py.readthedocs.io/en/stable/images.html#docker.models.images.ImageCollection.get) .